mPowerSDK/APHTasksReminderManager.m

```diff
@@ -0,0 +1,156 @@
+     if they have not already done tasks that day.
+     
+     - Should only remind them if they haven't done them yet today
+     - Reminder should randomly come in the periods that aren't marked as sleeping time
+     */
+    
+    
+    [self cancelLocalNotificationsIfExist];
+
+    if (self.reminderOn) {
+        
+        // We first must fetch tasks to see if any have been done today as we don't want to send
+        // a notification today if one or more have.
+        
+        NSDate *today = [NSDate date];
+        NSDate *yesterday = today.dayBefore;
+        
+        // creating and using a filter here eventhough we don't need one because we're looking for all
+        // the tasks. But, the fetch method in APCScheduler that does not require a filter is crashing
+        // because filter is nil, so it seem there's a bug there
+        
+        NSPredicate *filterForallTasks = [NSPredicate predicateWithFormat: @"%K == nil || %K == %@ || %K == %@",
+                                          NSStringFromSelector(@selector(taskIsOptional)),
+                                          NSStringFromSelector(@selector(taskIsOptional)),
+                                          @(NO),
+                                          NSStringFromSelector(@selector(taskIsOptional)),
+                                          @(YES)];
+
+        [[APCScheduler defaultScheduler] fetchTaskGroupsFromDate: yesterday
+                                                          toDate: today
+                                          forTasksMatchingFilter: filterForallTasks
+                                                      usingQueue: [NSOperationQueue mainQueue]
+                                                 toReportResults: ^(NSDictionary *taskGroups, NSError * __unused queryError)
+         {
+             NSArray *actualTaskGroups = taskGroups.allValues.firstObject;
+             if (actualTaskGroups.count > 0) {
+                 
+                 BOOL includeToday = YES;
+                 for (APCTaskGroup *taskGroup in actualTaskGroups) {
+                     if ([taskGroup.dateFullyCompleted.dayBefore isEqualToDate:[NSDate date].dayBefore]) {
+                         // same day
+                         includeToday = NO;
+                         break;
+                     }
+                 }
+                 
+                 [self scheduleDailyRandomReminderIncludeToday:includeToday];
+             }
+         }];  // first fetch:  required tasks, for a range of dates
+    }
+}
+
+- (void)scheduleDailyRandomReminderIncludeToday:(BOOL)includeToday {
+    
+    // Schedule the Task notification
+    UILocalNotification* taskNotification = [[UILocalNotification alloc] init];
+    taskNotification.alertBody = gTaskReminderMessage;
+    
+    NSDate* normalFireDate = [self randomDailyFireDate];
+    
+    // if the fire date is within an hour of current time, then skip for today
```

Comment:
While I grabbed an hour as a decent minimum time from now, I don't think it should be hardcoded down here. What about a variable name like minimumMinutesUntilReminder and then set it to 60? Preferably defined in an info.plist in an appropriate place.